src/compile.rs

```diff
@@ -207,6 +270,16 @@ fn compile_target(
         // but forwarding stderr is still useful in case there some non-json error
         .stderr(Stdio::inherit());
 
+    if let Some(zig_linker) = zig_linker {
+        build_command.env("CC", &zig_linker);
+        let env_target = context
+            .target
+            .target_triple()
+            .to_uppercase()
+            .replace("-", "_");
+        build_command.env(format!("CARGO_TARGET_{}_LINKER", env_target), &zig_linker);
```

Comment:
I think we should also set `$TARGET_CC` and maybe even `$TARGET_CXX` if zig cc supports it for the [cc](https://github.com/alexcrichton/cc-rs) crate in case of C/C++ source dependencies, for example [openssl-src](https://github.com/alexcrichton/openssl-src-rs).

Reply:
Do you have any example for where `TARGET_CC` is used? I couldn't find it searching the code of cc, cargo and openssl-src.

Reply:
It's here: https://github.com/alexcrichton/cc-rs/blob/388bf178f6c857bb07bcc333ceb31032fbee686c/src/lib.rs#L2356

The `get_var` method will try to find `<$TARGET>_CC`.